lib/puppet/util/selinux.rb

```diff
@@ -137,6 +142,33 @@ def set_selinux_default_context(file)
     nil
   end
 
+  ##
+  # normalize_selinux_category is an internal, method that converts
+  # all selinux categories to the human-readable representation,
+  # avoiding potential issues when mcstransd is not functional.
+  #
+  # It is not marked private because it is needed by File's
+  # selcontext.rb, but it is not intended for use outside of Puppet's
+  # code.
+  #
+  # @return [String] the human-readable category name
+  def normalize_selinux_category(category)
+    # We don't cache this, but there's already a ton of duplicate work
+    # in the selinux handling code.
+
+    File.open(Selinux.selinux_translations_path).each do |line|
```

Comment:
Are you worried about the performance implications of a system with many file resources with managed `:selrange` properties? I'm not sure if caching is a better idea.. The `chcat` source code doesn't appear to anyway: https://github.com/SELinuxProject/selinux/blob/7935dee8f62931159e3bdb7bc028c2fdb7ecd7da/python/chcat/chcat#L371-L380